src/azure-cli/azure/cli/command_modules/keyvault/_validators.py

```diff
@@ -374,3 +375,22 @@ def _validate(ns):
         attr_arg = attr_type(enabled=(not disabled))
         setattr(ns, attr_arg_name, attr_arg)
     return _validate
+
+
+def _extract_subresource_name_from_single_output(output, id_parameter):
+    if not getattr(output, id_parameter):
+        resource_name = None
+    else:
+        items = getattr(output, id_parameter).split('/')
+        resource_name = items[4] if len(items) > 4 else None
+
+    setattr(output, 'name', resource_name)
+    return output
+
+
+def extract_subresource_name(id_parameter='id'):
+    def _extract_subresource_name(output):
+        if isinstance(output, Iterable):
+            return [_extract_subresource_name_from_single_output(_, id_parameter) for _ in output]
```

Comment:
Underscore `_` usually means the value will be dropped. If it is used in another function call, we'd better give it some meaningful name.

Reply:
Really good point, renamed it to `item`.